src/yggdrasil/tcp.go

```diff
@@ -109,13 +142,35 @@ func (iface *tcpInterface) listener() {
 	iface.core.log.Println("Listening for TCP on:", iface.serv.Addr().String())
 	for {
 		sock, err := iface.serv.Accept()
-		if err != nil {
-			panic(err)
+		select {
+		case <-iface.serv_stop:
```

Comment:
Hm yes - I've added an error check again. I was wondering whether to go with `return` or `continue` in this case as I don't know if there might be cases where an `Accept` fails because of the remote side and not us and we would want the listener to continue running, but given that we were `panic()`ing before, `return` seems reasonable enough for now if you agree.

I don't really know how to discern between an error returned as a result of the `Accept` or the `Listener` itself, but if the initial `Listen` call fails then we never reach this point anyway.